core/services/job/models.go

```diff
@@ -918,15 +919,30 @@ func (w *WorkflowSpec) SDKSpec(ctx context.Context) (sdk.WorkflowSpec, error) {
 		return *w.sdkWorkflow, nil
 	}
 
-	spec, cid, err := workflowSpecFactory.Spec(ctx, w.Workflow, []byte(w.Config), w.SpecType)
+	spec, rawSpec, cid, err := workflowSpecFactory.Spec(ctx, w.Workflow, []byte(w.Config), w.SpecType)
```

Comment:
Why not pass the raw CID?  The point of passing spec instead of rawSpec was to save space, given that the compressed binary is much smaller.  If we pass it around anyways, we don't need the compressed one.

Since the binary is first in the sha256, you can do sum on just the spec, then do sum again after adding the bytes to git the cid.  It would also save compute during execution because you wouldn't need to generate the id each time.

Reply:
Hmmm... I might have misunderstood: rawSpec is be the compressed bytes, no?

Reply:
Ah ok, my change isn't in yet so I wasn't picturing it that way.